### PR TITLE
[Fix] Fix one_hot error about ignore_index in focalloss and bceloss

### PR DESCRIPTION
**Pull request by geoyee**
### PR types

Bug fixes

### PR changes

APIs

### Description

在`focalloss`和`bceloss`中未正确将label中由于部分几何变换（rotation等）填充的255去掉，因此在计算loss进行`one_hot`操作时报错如下：

```
Error: /paddle/paddle/phi/kernels/gpu/one_hot_kernel.cu:38 Assertion `p_in_data[idx] >= 0 && p_in_data[idx] < depth` failed. Illegal index value, Input(input) value should be greater than or equal to 0, and less than depth [2], but received [255].
```

此时数据检查无异常：

``` python
import numpy as np

for data in train_dataset:
    print(data["img"].shape, data["img"].dtype)
    print(data["label"].shape, data["label"].dtype, np.unique(data["label"]))
    break

# -- output --
# (3, 320, 480) float32
# (320, 480) uint8 [  0   1 255]
```

修复增加了`label = paddle.where(mask, label, paddle.zeros_like(label))`后，正常运行。

### CODE REVIEW

**Review by Bobholamovic — APPROVED** [review]
LGTM.
